diff -urN linux-4.1.23.orig/drivers/w1/slaves/Makefile linux-4.1.23/drivers/w1/slaves/Makefile
--- linux-4.1.23.orig/drivers/w1/slaves/Makefile	2016-04-25 17:57:36.000000000 +0200
+++ linux-4.1.23/drivers/w1/slaves/Makefile	2016-07-08 21:44:29.843301655 +0200
@@ -15,3 +15,4 @@
 obj-$(CONFIG_W1_SLAVE_DS2781)	+= w1_ds2781.o
 obj-$(CONFIG_W1_SLAVE_BQ27000)	+= w1_bq27000.o
 obj-$(CONFIG_W1_SLAVE_DS28E04)	+= w1_ds28e04.o
+obj-$(CONFIG_W1_SLAVE_DS28E17)	+= w1_ds28e17.o
